pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -222,14 +221,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 			conditionSynced := NewExternalSecretCondition(esv1beta1.ExternalSecretReady, v1.ConditionTrue, esv1beta1.ConditionReasonSecretDeleted, "secret deleted due to DeletionPolicy")
 			SetExternalSecretCondition(&externalSecret, *conditionSynced)
-			return ctrl.Result{RequeueAfter: requeueAfter}, nil
+			return ctrl.Result{}, err
```

Comment:
Wouldn't this one cause a secret that was deleted due to deletion policy to never sync again? I believe `err==nil` is possible in this path

Reply:
right, i've missed that!